### PR TITLE
Issue 194 wire events to place details

### PR DESCRIPTION
**Pull request by fluffyemily**
Detect when places have events and display their details

Put notifications / place details banner text into remote config so it can be tweaked without any need to redeploy the app

### CODE REVIEW

File: Prox/Prox/Models/Event.swift [hunk 1]
```diff
@@ -13,6 +14,48 @@ class Event {
...
+        return eventNotificationStrings
+    }()
+
+    var notificationString: String {
+        let randomIndex = Int(arc4random_uniform(UInt32(Event.numberOfEventNotificationStrings)))
+        return Event.eventNotificationStrings[randomIndex]
+    }
+
+    private static var numberOfPlaceDisplayStrings: Int = {
+        let key = RemoteConfigKeys.numberOfPlaceDetailsEventStrings
```
Comment (mcomella) [inline Prox/Prox/Models/Event.swift]: nit: it'd be nice to put the duplicated code in a helper fn, but to save time, wfm!

File: Prox/Prox/Models/Place.swift [hunk 2]
```diff
@@ -175,6 +175,44 @@ class Place: Hashable {
         }
     }
 
+    func getNotificationString(forEvent event: Event) -> String {
+        return replacePlaceName(string: replaceTimeToEvent(string: replaceEventName(string: event.notificationString, withName: event.description), withStartTime: event.startTime))
```
Comment (mcomella) [inline Prox/Prox/Models/Place.swift]: nit: pretty hard to read – it'd be easier with temp vars

File: Prox/Prox/Models/Place.swift [hunk 3]
```diff
@@ -175,6 +175,44 @@ class Place: Hashable {
...
+        return string.replacingOccurrences(of: "{venue_name}", with: self.name)
+    }
+
+    private func replaceTimeToEvent(string: String, withStartTime startTime: Date) -> String {
+        let now = Date()
+        let timeToEvent = now.timeIntervalSince(startTime)
+        let (hours, mins) = secondsToHoursMinutes(seconds: Int(timeToEvent))
+        var timeString: String = ""
+        if hours > 0 {
+            timeString += "\(hours) hours, "
```
Comment (mcomella) [inline Prox/Prox/Models/Place.swift]: nit: we'll get 1 hours

File: Prox/Prox/Models/Place.swift [hunk 4]
```diff
@@ -175,6 +175,44 @@ class Place: Hashable {
...
+        }
+        timeString += "\(mins) minutes"
```
Comment (mcomella) [inline Prox/Prox/Models/Place.swift]: nit: 1 minutes

**Comment by fluffyemily** [comment]
[Show events happening nearby](https://trello.com/c/Zf5kaT3F/195-show-events-happening-nearby)

**Pull request merged**